File: corporate-reporting/src/types/CustomReportingEntityType.ts
import { ICustomReportingEntity } from "./CustomReportingEntity";
import { IReportingField } from "./CustomReportingField";
import { Entity, IEntity } from "./Entity";
import { IReportType } from "./ReportType";

export interface ICustomReportingEntityType extends IEntity {
    ReportTypeID: number;
    InheritReportSchedule: boolean;
    IsHeadlineSection: boolean;
    UpdateHasRag: boolean;
    UpdateRagIsRequired: boolean;
    UpdateHasNarrative: boolean;
    UpdateNarrativeIsRequired: boolean;
    UpdateNarrativeMaxChars: number;
    UpdateHasDeliveryDates: boolean;
    UpdateDeliveryDatesIsRequired: boolean;
    HasUpperAndLowerTargets: boolean;
    UpdateHasMeasurement: boolean;
    UpdateMeasurementIsRequired: boolean;
    CustomFields: IReportingField[];

    ReportType?: IReportType;
    ReportingEntities?: ICustomReportingEntity[];
}

export class CustomReportingEntityType extends Entity implements ICustomReportingEntityType {
    public ReportTypeID = null;
    public InheritReportSchedule = false;
    public IsHeadlineSection = false;
    public UpdateHasRag = false;
    public UpdateRagIsRequired = false;
    public UpdateHasNarrative = false;
    public UpdateNarrativeIsRequired = false;
    public UpdateNarrativeMaxChars = 500;
    public UpdateHasDeliveryDates = false;
    public UpdateDeliveryDatesIsRequired = false;
    public HasUpperAndLowerTargets = false;
    public UpdateHasMeasurement = false;
    public UpdateMeasurementIsRequired = false;
    public CustomFields: IReportingField[] = [];
}

File: ORB.Core/Models/CustomReportingEntityType.cs
using System;
using System.Collections.Generic;

namespace ORB.Core.Models
{
    public class CustomReportingEntityType : Entity
    {
        public CustomReportingEntityType()
        {
            CustomFields = new HashSet<ReportingField>();
            ReportingEntities = new HashSet<CustomReportingEntity>();
        }

        public string Description { get; set; }
        public DateTime? CreatedDate { get; set; }
        public int ReportTypeID { get; set; }
        public bool? InheritReportSchedule { get; set; }
        public bool? IsHeadlineSection { get; set; }
        public bool? UpdateHasRag { get; set; }
        public bool? UpdateRagIsRequired { get; set; }
        public bool? UpdateHasNarrative { get; set; }
        public bool? UpdateNarrativeIsRequired { get; set; }
        public int? UpdateNarrativeMaxChars { get; set; }
        public bool? UpdateHasDeliveryDates { get; set; }
        public bool? UpdateDeliveryDatesIsRequired { get; set; }
        public bool? HasUpperAndLowerTargets { get; set; }
        public bool? UpdateHasMeasurement { get; set; }
        public bool? UpdateMeasurementIsRequired { get; set; }
        public ICollection<ReportingField> CustomFields { get; set; }

        public virtual ReportType ReportType { get; set; }
        public virtual ICollection<CustomReportingEntity> ReportingEntities { get; set; }
    }
}
